File: passgen-app/passgen-web/src/main/java/com/passgen/core/actors/FinActor.java
package com.passgen.core.actors;

import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import com.passgen.core.model.PasswordResult;

import static akka.actor.ActorRef.noSender;

public class FinActor extends UntypedActor {
    private final LoggingAdapter LOG = Logging.getLogger(getContext().system(), this);

    @Override
    public void onReceive(Object m) throws Exception {
        if (m instanceof PasswordResult) {
            LOG.info("Password result: {}", m);
            sender().tell(m, noSender());
        } else {
            unhandled(m);
        }
    }
}

File: passgen-app/passgen-scala/src/main/scala/com/passgen/core/actors/FinActor.scala
package com.passgen.core.actors

import akka.actor._
import akka.event.slf4j.SLF4JLogging
import com.passgen.core.model.PasswordResult

class FinActor extends Actor with SLF4JLogging {
  def receive = {
    case m: PasswordResult =>
      log.debug("Password result: {}", m)
      sender() ! m
  }
}